notebooks/07_add_dynamic_forcing.py

```diff
@@ -10,23 +11,57 @@
 endtime = datetime(2018, 1, 1)
 
 
-def add_forcing(model, cloud, starttime, endtime):
-    forcing = SetDynamicForcing(
-        model=model,
-        cloud=cloud,
-        startdate=starttime,
-        enddate=endtime,
-    )
-
-    model = forcing.add()
-
-    # Add dynamic groundwater
-    offline_budgets = AssignOfflineBudgets()
-    offline_budgets.compute_budgets(model)
+def add_forcing(model, cloud, starttime, endtime, cache_forcing: bool = False):
+    cache_file = model.filepath.parents[1] / "basin_time.arrow"
```

Comment:
Can you add a docstring that explains what `cache_forcing` does?

This `cache_file` if it is real input, it should be relative to the `input_dir` (now defaults to "input" instead of "."), and you can use `model.basin.time.filepath` as well, this is set for all `ribasim_nl.Model`.